diff --git a/src/third_party/skia/third_party/skcms/src/Transform_inl.h b/src/third_party/skia/third_party/skcms/src/Transform_inl.h
index cb11777..2d994b4 100644
--- a/src/third_party/skia/third_party/skcms/src/Transform_inl.h
+++ b/src/third_party/skia/third_party/skcms/src/Transform_inl.h
@@ -456,7 +456,9 @@ SI ATTR F NS(F_from_U8_)(U8 v) {
 SI ATTR F NS(F_from_U16_BE_)(U16 v) {
     // All 16-bit ICC values are big-endian, so we byte swap before converting to float.
     // MSVC catches the "loss" of data here in the portable path, so we also make sure to mask.
-    v = (U16)( ((v<<8)|(v>>8)) & 0xffff );
+    U16 t1 = (v<<8) & 0xff00;
+    U16 t2 = (v>>8) & 0x00ff;
+    v = (U16) t1 | t2;
     return CAST(F, v) * (1/65535.0f);
 }
 #define F_from_U16_BE NS(F_from_U16_BE_)
